spark3/src/test/java/org/apache/iceberg/spark/source/TestStructuredStreamingRead3.java

```diff
@@ -0,0 +1,435 @@
+
+    table.refresh();
+
+    Dataset<Row> ds = spark.readStream()
+        .format("iceberg")
+        .load(tableIdentifier);
+    Assertions.assertThat(processAvailable(ds))
+        .containsExactlyInAnyOrderElementsOf(Iterables.concat(parquetFileRecords, orcFileRecords, avroFileRecords));
+  }
+
+  @SuppressWarnings("unchecked")
+  @Test
+  public void testReadStreamFromEmptyTable() throws Exception {
+    table.refresh();
+
+    Dataset<Row> df = spark.readStream()
+        .format("iceberg")
+        .load(tableIdentifier);
+
+    List<SimpleRecord> actual = processAvailable(df);
+    Assert.assertEquals(Collections.emptyList(), actual);
+  }
+
+  @SuppressWarnings("unchecked")
+  @Test
+  public void testReadStreamWithSnapshotTypeOverwriteErrorsOut() throws Exception {
+    // upgrade table to version 2 - to facilitate creation of Snapshot of type OVERWRITE.
+    TableOperations ops = ((BaseTable) table).operations();
+    TableMetadata meta = ops.current();
+    ops.commit(meta, meta.upgradeToFormatVersion(2));
+
+    // fill table with some initial data
+    List<List<SimpleRecord>> dataAcrossSnapshots = TEST_DATA_MULTIPLE_SNAPSHOTS;
+    appendDataAsMultipleSnapshots(dataAcrossSnapshots, tableIdentifier);
+
+    Schema deleteRowSchema = table.schema().select("data");
+    Record dataDelete = GenericRecord.create(deleteRowSchema);
+    List<Record> dataDeletes = Lists.newArrayList(
+        dataDelete.copy("data", "one") // id = 1
+    );
+
+    DeleteFile eqDeletes = FileHelpers.writeDeleteFile(
+        table, Files.localOutput(temp.newFile()), TestHelpers.Row.of(0), dataDeletes, deleteRowSchema);
+
+    table.newRowDelta()
+        .addDeletes(eqDeletes)
+        .commit();
+
+    // check pre-condition - that the above Delete file write - actually resulted in snapshot of type OVERWRITE
+    Assert.assertEquals(DataOperations.OVERWRITE, table.currentSnapshot().operation());
+
+    Dataset<Row> df = spark.readStream()
+        .format("iceberg")
+        .load(tableIdentifier);
+    StreamingQuery streamingQuery = df.writeStream()
+        .format("memory")
+        .queryName("testtablewithoverwrites")
+        .outputMode(OutputMode.Append())
+        .start();
+
+    AssertHelpers.assertThrowsCause(
+        "Streaming should fail with IllegalStateException, as the snapshot is not of type APPEND",
+        IllegalStateException.class,
+        "Invalid Snapshot operation",
+        () -> streamingQuery.processAllAvailable()
+    );
+  }
+
+  @SuppressWarnings("unchecked")
+  @Test
+  public void testReadStreamWithSnapshotTypeReplaceErrorsOut() throws Exception {
+    // fill table with some data
+    List<List<SimpleRecord>> dataAcrossSnapshots = TEST_DATA_MULTIPLE_SNAPSHOTS;
+    appendDataAsMultipleSnapshots(dataAcrossSnapshots, tableIdentifier);
+
+    table.refresh();
+
+    // this should create a snapshot with type Replace.
+    table.rewriteManifests()
+        .clusterBy(f -> 1)
+        .commit();
```

Comment:
It should always be safe to simply skip `replace` commits because they should never change the data in the table, just the data files or manifests. It's fine to fail for now, but I think you'd want to support skipping these pretty quickly so you can support compaction in the background.

Reply:
yes, this is on the top of my **What's Next** list. A spark option to `ignore-replace`.